3-Network-Configuration/3.5-Firewall-Configuration.bats

```diff
@@ -0,0 +1,244 @@
+
+# 3.5.1 Configure UncomplicatedFirewall
+
+@test "3.5.1.1 Ensure ufw is installed (Automated)" {
+    [[ $(dpkg -s ufw | grep 'Status: install') == "Status: install ok installed" ]]
+}
+
+@test "3.5.1.2 Ensure iptables-persistent is not installed with ufw (Automated)" {
+    run dpkg-query -s iptables-persistent
+    [ $status -eq 1 ]
+    [[ "$output" == *"package 'iptables-persistent' is not installed and no information is available"* ]]
+}
+
+@test "3.5.1.3 Ensure ufw service is enabled (Automated)" {
+    [[ $(systemctl is-enabled ufw) == "enabled" ]]
+    [[ $(ufw status | grep Status) == "Status: active" ]]
+}
+
+@test "3.5.1.4 Ensure ufw loopback traffic is configured (Automated)" {
+    run bash -c "ufw status verbose"
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"Anywhere on lo"*"ALLOW IN"*"Anywhere"* ]]
+    [[ "$output" == *"Anywhere"*"DENY IN"*"127.0.0.0/8"* ]]
+    [[ "$output" == *"Anywhere (v6) on lo"*"ALLOW IN"*"Anywhere (v6)"* ]]
+    [[ "$output" == *"Anywhere (v6)"*"DENY IN"*"::1"* ]]
+    [[ "$output" == *"Anywhere"*"ALLOW OUT"*"Anywhere on lo"* ]]
+    [[ "$output" == *"Anywhere (v6)"*"ALLOW OUT"*"Anywhere (v6) on lo"* ]]
+}
+
+@test "3.5.1.5 Ensure ufw outbound connections are configured (Manual)" {
+    skip "This audit has to be done manually"
+}
+
+@test "3.5.1.6 Ensure ufw firewall rules exist for all open ports (Manual)" {
+    skip "This audit has to be done manually"
+}
+
+@test "3.5.1.7 Ensure ufw default deny firewall policy (Automated)" {
+    run bash -c "ufw status verbose | grep -i default"
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"deny (incoming)"* || "$output" == *"reject (incoming)"* ]]
+    [[ "$output" == *"deny (outgoing)"* || "$output" == *"reject (outgoing)"* ]]
+    [[ "$output" == *"deny (routed)"* \
+     || "$output" == *"reject (routed)"* \
+     || "$output" == *"disabled (routed)"* ]]
+}
+
+# 3.5.2 Configure nftables
+
+@test "3.5.2.1 Ensure nftables is installed (Automated)" {
+    run bash -c "dpkg -s nftables | grep 'Status: install'"
+    [[ "$output" == "Status: install ok installed" ]]
+}
+
+@test "3.5.2.2 Ensure ufw is uninstalled or disabled with nftables (Automated)" {
+    run bash -c "dpkg -s ufw | grep 'dpkg-query'"
+    if [[ ! "$output" == *"dpkg-query: package 'ufw' is not installed and no information is available"* ]]; then
+        [[ $(ufw status | grep 'Status') == "Status: inactive" ]]
+    fi
+}
+
+@test "3.5.2.3 Ensure iptables are flushed with nftables (Manual)" {
+    run bash -c "iptables -L"
+    [ "$status" -eq 0 ]
+    [ "$output" = "" ]
+    run bash -c "ip6tables -L"
+    [ "$status" -eq 0 ]
+    [ "$output" = "" ]
+}
+
+@test "3.5.2.4 Ensure a nftables table exists (Automated)" {
+    run bash -c "nft list tables"
+    [ "$status" -eq 0 ]
+}
+
+@test "3.5.2.5 Ensure nftables base chains exist (Automated)" {
+    run bash -c "nft list ruleset | grep 'hook input'"
+    [ "$status" -eq 0 ]
+    [[ "$output" == *"type filter hook input priority filter;"* ]]
```

Comment:
This does not match the line from the CIS document (`type filter hook input priority 0;`)
Also check the other two in this test

Reply:
This must be an error in the CIS document. `priority 0` will be outputted as `priority filter`.
Here you can see a list of possible priority settings: https://wiki.nftables.org/wiki-nftables/index.php?title=Configuring_chains&oldid=208#Base_chain_priority

Reply:
Yes, it seems so. ok.